docs/essentials/Collections and Data Structures.md

```diff
@@ -0,0 +1,79 @@
+# Collections and Data Structures
+
+Similar data can often be handled more efficiently when stored and manipulated as a collection. You can use the [System.Array](http://dotnet.github.io/api/System.Array.html) class or the classes in the [System.Collections](http://dotnet.github.io/api/System.Collections), [System.Collections.Generic](http://dotnet.github.io/api/System.Collections.Generic.html), or [System.Collections.Concurrent](http://dotnet.github.io/api/System.Collections.Concurrent.html) namespaces to add, remove, and modify either individual elements or a range of elements in a collection.
+
+There are two main types of collections; generic collections and non-generic collections. Generic collections are type-safe at compile time. Because of this, generic collections typically offer better performance. Generic collections accept a type parameter when they are constructed and do not require that you cast to and from the [Object](http://dotnet.github.io/api/System.Object.html) type when you add or remove items from the collection. Non-generic collections store items as [Object](http://dotnet.github.io/api/System.Object.html) and require casting. You may see non-generic collections in older code.
+
+Starting with the .NET Framework 4, the collections in the [System.Collections.Concurrent](http://dotnet.github.io/api/System.Collections.Concurrent.html) namespace provide efficient thread-safe operations for accessing collection items from multiple threads.
```

Comment:
I'd remove anything that talks about the framework here since .NET Core already comes with that collection